lib/logstash/inputs/http.rb

```diff
@@ -177,12 +182,50 @@ def decode_body(headers, remote_address, body, default_codec, additional_codecs)
   end
 
   def push_decoded_event(headers, remote_address, event)
+    add_ecs_fields(headers, event)
     event.set(@request_headers_target_field, headers)
     event.set(@remote_host_target_field, remote_address)
     decorate(event)
     @queue << event
   end
 
+  def add_ecs_fields(headers, event)
+    return if ecs_compatibility == :disabled
+
+    http_version = headers.get("http_version")
+    event.set("[http][version]", http_version) if http_version
+
+    http_user_agent = headers.get("http_user_agent")
+    event.set("[user_agent][original]", http_user_agent) if http_user_agent
+
+    http_host = headers.get("http_host")
+    domain, port = self.class.get_domain_port(http_host)
```

Comment:
I think you could avoid `self.class` if a method is not found in instance, Ruby searches in the class object

Reply:
`self.class` is required for class method